packages/edit-site/src/components/block-editor/block-editor-provider/default-block-editor-provider.js

```diff
@@ -0,0 +1,118 @@
+
+/**
+ * Internal dependencies
+ */
+import { store as editSiteStore } from '../../../store';
+import { unlock } from '../../../lock-unlock';
+import useSiteEditorSettings from '../use-site-editor-settings';
+import { PAGE_CONTENT_BLOCK_TYPES } from '../../page-content-focus-manager/constants';
+
+const { ExperimentalBlockEditorProvider } = unlock( blockEditorPrivateApis );
+
+const noop = () => {};
+
+/**
+ * The default block editor provider for the site editor. Typically used when
+ * the post type is `'wp_template_part'` or `'wp_template'` and allows editing
+ * of the template and its nested entities.
+ *
+ * If the page content focus type is `'hideTemplate'`, the provider will provide
+ * a set of "ghost" blocks that mimick the look and feel of the post editor and
+ * allow editing of the page content only.
+ *
+ * @param {Object}    props
+ * @param {WPElement} props.children
+ */
+export default function DefaultBlockEditorProvider( { children } ) {
+	const settings = useSiteEditorSettings();
+
+	const { templateType, pageContentFocusType } = useSelect( ( select ) => {
+		const { getEditedPostType, getPageContentFocusType } =
+			select( editSiteStore );
+
+		return {
+			templateType: getEditedPostType(),
+			pageContentFocusType: getPageContentFocusType(),
+		};
+	}, [] );
+
+	const [ blocks, onInput, onChange ] = useEntityBlockEditor(
+		'postType',
+		templateType
+	);
+	const pageGhostBlocks = usePageContentBlocks( blocks );
+	const isTemplateHidden = pageContentFocusType === 'hideTemplate';
+
+	return (
+		<ExperimentalBlockEditorProvider
+			settings={ settings }
+			value={ isTemplateHidden ? pageGhostBlocks : blocks }
+			onInput={ isTemplateHidden ? noop : onInput }
+			onChange={ isTemplateHidden ? noop : onChange }
+			useSubRegistry={ false }
+		>
+			{ children }
+		</ExperimentalBlockEditorProvider>
+	);
+}
+
+/**
+ * Helper method to iterate through all blocks, recursing into inner blocks.
+ * Returns a flattened object.
+ *
+ * @param {Array} blocks Blocks to flatten.
+ *
+ * @return {Array} Flattened object.
+ */
+function flattenBlocks( blocks ) {
+	const result = [];
+
+	blocks.forEach( ( block ) => {
+		result.push( block );
+		result.push( ...flattenBlocks( block.innerBlocks ) );
+	} );
+
+	return result;
+}
+
+/**
+ * Returns a memoized array of blocks that contain only page content blocks,
+ * surrounded by a group block to mimic the post editor.
+ *
+ * @param {Array} blocks Block list.
+ *
+ * @return {Array} Flattened object.
+ */
+function usePageContentBlocks( blocks ) {
+	return useMemo( () => {
+		if ( ! blocks || ! blocks.length ) {
+			return [];
+		}
+		return [
+			createBlock(
+				'core/group',
+				{
+					layout: { type: 'constrained' },
+					style: {
+						spacing: {
+							margin: {
+								top: '4em', // Mimics the post editor.
+							},
+						},
+					},
+				},
+				flattenBlocks( blocks ).filter( ( block ) =>
+					PAGE_CONTENT_BLOCK_TYPES.includes( block.name )
+				)
```

Comment:
> In the commits prior to https://github.com/WordPress/gutenberg/commit/69deb7d10ebf9d0a32893345f36776929fef1d15, I think we were only creating one of each block, matched by name?

Oh yeah, in my testing I forgot to revert back to createBlock for one of each page types. 😅 

I'll take another look based on your research, thanks again!

Good to hear that it's actually rendering something useful now though!

Reply:
I've reverted back to using `createBlock`. Also the while loop/stack has a funny side effect of reversing the order of found objects even using `pop()`, so I've opted for a basic, but reliable for loop + continue. 

I think it works better, and have tested it with `core/query` as well as multiples of page content blocks. I'm going to write some unit tests tomorrow to increase my confidence. Right now, you are my unit tests and that's not fair on you 🤣

Reply:
> Also the while loop/stack has a funny side effect of reversing the order of found objects even using pop(), so I've opted for a basic, but reliable for loop + continue.

Ah, it possibly would need `stack.push` to be `stack.unshift` instead for it to be a depth-first traversal, otherwise siblings might be accessed before inner blocks as inner blocks get placed on the end of the stack? No matter, though, the `for` loop with `continue` looks fine to me 🙂

Reply:
> Right now, you are my unit tests and that's not fair on you 🤣

Always happy to be a unit test, the only problem is that I'm non-deterministic 😆

Reply:
> otherwise siblings might be accessed before inner blocks as inner blocks get placed on the end of the stack?

At that point I thought it might have created for both me and others a bit of order-related mind cluster bomb, so I thought the for loop was neater for the human brain 😄

Reply:
> so I thought the for loop was neater for the human brain 

Totally, better not to include "code test"-style examples / brain teasers in the code base unless necessary. `for` loops are much easier to read IMO, too 🙂